### ISSUE TITLE
Memory Corruption crash objc_retain in [ADB_HB_Channel request:]

### ISSUE THREAD

**Issue by nuno-vieira** [Adobe-Marketing-Cloud/media-sdks#60]
There's seems to be a crash in [ADB_HB_Channel request:] related to some kind of memory corruption. Here's a hint from crashlytics of what could be the cause of the crash:

> The stack trace indicates that heap corruption may have caused your app to crash. Memory corruption can occur pretty easily from freeing a dangling pointer, a thread race or bad pointer arithmetic. The important thing to keep in mind is that the resulting crash may happen long after the initial corruption. As a result, the stack trace for this crash might not provide any clues to the location of the bug in your code. However, you can still fix memory issues with tools from Apple. For speedy resolution of memory corruption issues, we recommend regularly auditing your app with Xcode's memory debugging facilities: Visual Memory Debugger, Zombies Instrument, Address Sanitizer, Thread Sanitizer and malloc diagnostics.

We were not able to reproduce this issue for now, it is only be reported in the crash report of our costumers. 
Let me know if any additional information is needed, and if you require a stack trace of the other threads.

The stack trace is kinda similar to this one: https://github.com/Adobe-Marketing-Cloud/media-sdks/issues/51
And we are already using the latest adobe version, 2.2.7.

Stack Trace:
```
Crashed: com.adobe.analytics.heartbeat.work.report (QOS: UNSPECIFIED)
0  libobjc.A.dylib                0x1c4829ca0 objc_retain + 16
1  AddonManager                   0x1063842bc -[ADB_HB_Channel request:] + 48
2  AddonManager                   0x1063864f4 -[ADB_HB_ReportFilter _flushBufferReport:] + 1000
3  AddonManager                   0x106388390 __28-[ADB_HB_ReportFilter flush]_block_invoke + 56
4  AddonManager                   0x10639140c -[ADB_HB_Callback call:] + 76
5  AddonManager                   0x1063914d4 -[ADB_HB_Callback apply:] + 148
6  AddonManager                   0x106390604 -[ADB_HB_Command run] + 88
7  AddonManager                   0x1063930a0 __34-[ADB_HB_CommandQueue addCommand:]_block_invoke33 + 84
8  Foundation                     0x1b2282cc8 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
9  Foundation                     0x1b217fb18 -[NSBlockOperation main] + 100
10 Foundation                     0x1b2284f80 __NSOPERATION_IS_INVOKING_MAIN__ + 20
11 Foundation                     0x1b217f7d4 -[NSOperation start] + 784
12 Foundation                     0x1b2285978 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 20
13 Foundation                     0x1b2285444 __NSOQSchedule_f + 180
14 libdispatch.dylib              0x1b0bce93c _dispatch_block_async_invoke2 + 104
15 libdispatch.dylib              0x1b0c1f280 _dispatch_client_callout + 16
16 libdispatch.dylib              0x1b0bc456c _dispatch_continuation_pop$VARIANT$mp + 412
17 libdispatch.dylib              0x1b0bc3cb0 _dispatch_async_redirect_invoke + 596
18 libdispatch.dylib              0x1b0bd111c _dispatch_root_queue_drain + 376
19 libdispatch.dylib              0x1b0bd18e4 _dispatch_worker_thread2 + 124
20 libsystem_pthread.dylib        0x1f5b54568 _pthread_wqthread + 212
21 libsystem_pthread.dylib        0x1f5b57874 start_wqthread + 8
```

**Comment by praveek** [comment]
Thanks for report @nuno-vieira. We will investigate it. How frequent is this crash ?

**Comment by nuno-vieira** [comment]
> Thanks for report @nuno-vieira. We will investigate it. How frequent is this crash ? It will be helpful if you can share the call stack of other threads.

@praveek Currently in 5 days this has caused 702 crashes affecting 696 users. So I would say it is not very frequent, but is affecting already some amount of users.

I'll get the full stack trace in the mean time.

**Comment by nuno-vieira** [comment]
Full Stack Trace: (Btw this is Firebase stack trace)

```
Crashed: com.adobe.analytics.heartbeat.work.report (QOS: UNSPECIFIED)
EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x0000000bb1a429b0

Crashed: com.adobe.analytics.heartbeat.work.report (QOS: UNSPECIFIED)
0  libobjc.A.dylib                0x186085980 objc_retain + 16
1  AddonManager                   0x103fc82bc -[ADB_HB_Channel request:] + 48
2  AddonManager                   0x103fca4f4 -[ADB_HB_ReportFilter _flushBufferReport:] + 1000
3  AddonManager                   0x103fcc390 __28-[ADB_HB_ReportFilter flush]_block_invoke + 56
4  AddonManager                   0x103fd540c -[ADB_HB_Callback call:] + 76
5  AddonManager                   0x103fd54d4 -[ADB_HB_Callback apply:] + 148
6  AddonManager                   0x103fd4604 -[ADB_HB_Command run] + 88
7  AddonManager                   0x103fd70a0 __34-[ADB_HB_CommandQueue addCommand:]_block_invoke33 + 84
8  Foundation                     0x18671d3f0 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
9  Foundation                     0x186628710 -[NSBlockOperation main] + 84
10 Foundation                     0x18671f64c __NSOPERATION_IS_INVOKING_MAIN__ + 20
11 Foundation                     0x186628414 -[NSOperation start] + 740
12 Foundation                     0x186720044 __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 20
13 Foundation                     0x18671fb10 __NSOQSchedule_f + 180
14 libdispatch.dylib              0x185fc0c8c _dispatch_block_async_invoke2 + 104
15 libdispatch.dylib              0x18600e524 _dispatch_client_callout + 16
16 libdispatch.dylib              0x185fb6e98 _dispatch_continuation_pop$VARIANT$mp + 412
17 libdispatch.dylib              0x185fb65ec _dispatch_async_redirect_invoke + 656
18 libdispatch.dylib              0x185fc3438 _dispatch_root_queue_drain + 376
19 libdispatch.dylib              0x185fc3bf8 _dispatch_worker_thread2 + 124
20 libsystem_pthread.dylib        0x18605fb38 _pthread_wqthread + 212
21 libsystem_pthread.dylib        0x186062740 start_wqthread + 8com.apple.main-thread


com.apple.main-thread
0  libsystem_kernel.dylib         0x18611c198 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x18611b60c mach_msg + 72
2  CoreFoundation                 0x1862c6328 __CFRunLoopServiceMachPort + 148
3  CoreFoundation                 0x1862c135c __CFRunLoopRun + 1160
4  CoreFoundation                 0x1862c0ba8 CFRunLoopRunSpecific + 424
5  GraphicsServices               0x190437344 GSEventRunModal + 160
6  UIKitCore                      0x18a3fc3e4 UIApplicationMain + 1932
7  PeacockMobile-US               0x10274b340 main + 4337414976 (&lt;compiler-generated&gt;:4337414976)
8  libdyld.dylib                  0x1861488f0 start + 4com.apple.uikit.eventfetch-thread


com.apple.uikit.eventfetch-thread
0  libsystem_kernel.dylib         0x18611c198 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x18611b60c mach_msg + 72
2  CoreFoundation                 0x1862c6328 __CFRunLoopServiceMachPort + 148
3  CoreFoundation                 0x1862c135c __CFRunLoopRun + 1160
4  CoreFoundation                 0x1862c0ba8 CFRunLoopRunSpecific + 424
5  Foundation                     0x18661201c -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 228
6  Foundation                     0x186611efc -[NSRunLoop(NSRunLoop) runUntilDate:] + 88
7  UIKitCore                      0x18a49f57c -[UIEventFetcher threadMain] + 152
8  Foundation                     0x186740e20 __NSThread__start__ + 848
9  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
10 libsystem_pthread.dylib        0x18606274c thread_start + 8com.google.firebase.crashlytics.MachExceptionServer


com.google.firebase.crashlytics.MachExceptionServer
0  PeacockMobile-US               0x102b21c1c FIRCLSProcessRecordAllThreads + 392 (FIRCLSProcess.c:392)
1  PeacockMobile-US               0x102b22000 FIRCLSProcessRecordAllThreads + 423 (FIRCLSProcess.c:423)
2  PeacockMobile-US               0x102b18914 FIRCLSHandler + 34 (FIRCLSHandler.m:34)
3  PeacockMobile-US               0x102b1b250 FIRCLSMachExceptionServer + 524 (FIRCLSMachException.c:524)
4  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
5  libsystem_pthread.dylib        0x18606274c thread_start + 8com.facebook.react.JavaScript


com.facebook.react.JavaScript
0  JavaScriptCore                 0x19522464c WTF::Thread::addToThreadGroup(WTF::AbstractLocker const&amp;, WTF::ThreadGroup&amp;) + 48
1  JavaScriptCore                 0x19522390c WTF::ThreadGroup::addCurrentThread() + 116
2  JavaScriptCore                 0x19522390c WTF::ThreadGroup::addCurrentThread() + 116
3  JavaScriptCore                 0x195c99600 JSC::JSLock::didAcquireLock() + 204
4  JavaScriptCore                 0x195c99028 JSC::JSLockHolder::JSLockHolder(JSC::VM&amp;) + 52
5  JavaScriptCore                 0x1954a0484 JSObjectGetPropertyAtIndex + 60
6  PeacockMobile-US               0x102c5cb10 facebook::jsc::JSCRuntime::getValueAtIndex(facebook::jsi::Array const&amp;, unsigned long) + 977 (JSCRuntime.cpp:977)
7  PeacockMobile-US               0x102c61a30 facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1006 (jsi.h:1006)
8  PeacockMobile-US               0x102c61884 facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1018 (dynamic-inl.h:1018)
9  PeacockMobile-US               0x102c61b0c facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1018 (dynamic-inl.h:1018)
10 PeacockMobile-US               0x102c61884 facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1018 (dynamic-inl.h:1018)
11 PeacockMobile-US               0x102c61b0c facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1018 (dynamic-inl.h:1018)
12 PeacockMobile-US               0x102c61b0c facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1018 (dynamic-inl.h:1018)
13 PeacockMobile-US               0x102c61884 facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1018 (dynamic-inl.h:1018)
14 PeacockMobile-US               0x102c61884 facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1018 (dynamic-inl.h:1018)
15 PeacockMobile-US               0x102c61884 facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1018 (dynamic-inl.h:1018)
16 PeacockMobile-US               0x102c61884 facebook::jsi::dynamicFromValue(facebook::jsi::Runtime&amp;, facebook::jsi::Value const&amp;) + 1018 (dynamic-inl.h:1018)
17 PeacockMobile-US               0x102c637cc facebook::react::JSIExecutor::callNativeModules(facebook::jsi::Value const&amp;, bool) + 307 (JSIExecutor.cpp:307)
18 PeacockMobile-US               0x102c63488 facebook::react::JSIExecutor::callFunction(std::__1::basic_string&lt;char, std::__1::char_traits&lt;char&gt;, std::__1::allocator&lt;char&gt; &gt; const&amp;, std::__1::basic_string&lt;char, std::__1::char_traits&lt;char&gt;, std::__1::allocator&lt;char&gt; &gt; const&amp;, folly::dynamic const&amp;) + 228 (JSIExecutor.cpp:228)
19 PeacockMobile-US               0x102c595b8 std::__1::__function::__func&lt;facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function&lt;void (facebook::react::JSExecutor*)&gt;)::$_7, std::__1::allocator&lt;facebook::react::NativeToJsBridge::runOnExecutorQueue(std::__1::function&lt;void (facebook::react::JSExecutor*)&gt;)::$_7&gt;, void ()&gt;::operator()() + 1707 (functional:1707)
20 PeacockMobile-US               0x102bc13b8 facebook::react::tryAndReturnError(std::__1::function&lt;void ()&gt; const&amp;) + 4342092728 (RCTCxxUtils.mm:4342092728)
21 PeacockMobile-US               0x102bcef14 facebook::react::RCTMessageThread::tryFunc(std::__1::function&lt;void ()&gt; const&amp;) + 59 (RCTMessageThread.mm:59)
22 CoreFoundation                 0x1862c690c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 20
23 CoreFoundation                 0x1862c6098 __CFRunLoopDoBlocks + 264
24 CoreFoundation                 0x1862c16c4 __CFRunLoopRun + 2032
25 CoreFoundation                 0x1862c0ba8 CFRunLoopRunSpecific + 424
26 PeacockMobile-US               0x102bb6e00 +[RCTCxxBridge runRunLoop] + 268 (RCTCxxBridge.mm:268)
27 Foundation                     0x186740e20 __NSThread__start__ + 848
28 libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
29 libsystem_pthread.dylib        0x18606274c thread_start + 8JavaScriptCore bmalloc scavenger


JavaScriptCore bmalloc scavenger
0  libsystem_kernel.dylib         0x18613db58 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x186057ce4 _pthread_cond_wait$VARIANT$mp + 680
2  libc++.1.dylib                 0x18618f7ac std::__1::condition_variable::wait(std::__1::unique_lock&lt;std::__1::mutex&gt;&amp;) + 24
3  JavaScriptCore                 0x195261904 void std::__1::condition_variable_any::wait&lt;std::__1::unique_lock&lt;bmalloc::Mutex&gt; &gt;(std::__1::unique_lock&lt;bmalloc::Mutex&gt;&amp;) + 108
4  JavaScriptCore                 0x1952655dc bmalloc::Scavenger::threadRunLoop() + 160
5  JavaScriptCore                 0x1952652ec bmalloc::Scavenger::Scavenger(std::__1::lock_guard&lt;bmalloc::Mutex&gt; const&amp;) + 10
6  JavaScriptCore                 0x195266530 std::__1::__thread_specific_ptr&lt;std::__1::__thread_struct&gt;::set_pointer(std::__1::__thread_struct*) + 38
7  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
8  libsystem_pthread.dylib        0x18606274c thread_start + 8com.apple.CoreMotion.MotionThread


com.apple.CoreMotion.MotionThread
0  libsystem_kernel.dylib         0x18611c198 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x18611b60c mach_msg + 72
2  CoreFoundation                 0x1862c6328 __CFRunLoopServiceMachPort + 148
3  CoreFoundation                 0x1862c135c __CFRunLoopRun + 1160
4  CoreFoundation                 0x1862c0ba8 CFRunLoopRunSpecific + 424
5  CoreFoundation                 0x1862c1904 CFRunLoopRun + 60
6  CoreMotion                     0x192f556f4 CLClientCreateIso6709Notation + 150064
7  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
8  libsystem_pthread.dylib        0x18606274c thread_start + 8com.apple.NSURLConnectionLoader


com.apple.NSURLConnectionLoader
0  libsystem_kernel.dylib         0x18611c198 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x18611b60c mach_msg + 72
2  CoreFoundation                 0x1862c6328 __CFRunLoopServiceMachPort + 148
3  CoreFoundation                 0x1862c135c __CFRunLoopRun + 1160
4  CoreFoundation                 0x1862c0ba8 CFRunLoopRunSpecific + 424
5  CFNetwork                      0x18958e5d0 (Missing)
6  Foundation                     0x186740e20 __NSThread__start__ + 848
7  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
8  libsystem_pthread.dylib        0x18606274c thread_start + 8AVAudioSession Notify Thread


AVAudioSession Notify Thread
0  libsystem_kernel.dylib         0x18611c198 mach_msg_trap + 8
1  libsystem_kernel.dylib         0x18611b60c mach_msg + 72
2  CoreFoundation                 0x1862c6328 __CFRunLoopServiceMachPort + 148
3  CoreFoundation                 0x1862c135c __CFRunLoopRun + 1160
4  CoreFoundation                 0x1862c0ba8 CFRunLoopRunSpecific + 424
5  AVFAudio                       0x193097a2c GenericRunLoopThread::Entry(void*) + 156
6  AVFAudio                       0x1930e87a0 CAPThread::Entry(CAPThread*) + 204
7  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
8  libsystem_pthread.dylib        0x18606274c thread_start + 8ComScore-Thread


ComScore-Thread
0  libsystem_kernel.dylib         0x18613db58 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x186057d18 _pthread_cond_wait$VARIANT$mp + 732
2  AddonManager                   0x1040b1824 ComScore::WaitableEvent::wait(int) const + 188
3  AddonManager                   0x1040fce7c ComScore::ContextTasksThread&lt;ComScore::Core&gt;::run() + 224
4  AddonManager                   0x1040b1ccc ComScore::Thread::threadEntryPoint() + 104
5  AddonManager                   0x1040b3270 threadEntryProc + 32
6  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
7  libsystem_pthread.dylib        0x18606274c thread_start + 8ComScore-NetworkThread


ComScore-NetworkThread
0  libsystem_kernel.dylib         0x18613db58 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x186057d18 _pthread_cond_wait$VARIANT$mp + 732
2  AddonManager                   0x1040b1824 ComScore::WaitableEvent::wait(int) const + 188
3  AddonManager                   0x1040fce7c ComScore::ContextTasksThread&lt;ComScore::Core&gt;::run() + 224
4  AddonManager                   0x1040b1ccc ComScore::Thread::threadEntryPoint() + 104
5  AddonManager                   0x1040b3270 threadEntryProc + 32
6  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
7  libsystem_pthread.dylib        0x18606274c thread_start + 8ComScore-StreamingAnalytics-1


ComScore-StreamingAnalytics-1
0  libsystem_kernel.dylib         0x18613db58 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x186057d18 _pthread_cond_wait$VARIANT$mp + 732
2  AddonManager                   0x1040b1824 ComScore::WaitableEvent::wait(int) const + 188
3  AddonManager                   0x1041057a8 ComScore::ContextTasksThread&lt;ComScore::StreamingCore&gt;::run() + 224
4  AddonManager                   0x1040b1ccc ComScore::Thread::threadEntryPoint() + 104
5  AddonManager                   0x1040b3270 threadEntryProc + 32
6  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
7  libsystem_pthread.dylib        0x18606274c thread_start + 8Thread #1


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #2


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #3


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #4


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8JSC Heap Collector Thread


JSC Heap Collector Thread
0  libsystem_kernel.dylib         0x18613db58 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x186057ce4 _pthread_cond_wait$VARIANT$mp + 680
2  JavaScriptCore                 0x195226500 WTF::ThreadCondition::timedWait(WTF::Mutex&amp;, WTF::WallTime) + 152
3  JavaScriptCore                 0x19520b13c WTF::ParkingLot::parkConditionallyImpl(void const*, WTF::ScopedLambda&lt;bool ()&gt; const&amp;, WTF::ScopedLambda&lt;void ()&gt; const&amp;, WTF::TimeWithDynamicClockType const&amp;) + 1908
4  JavaScriptCore                 0x1951e2d18 bool WTF::Condition::waitUntil&lt;WTF::Lock&gt;(WTF::Lock&amp;, WTF::TimeWithDynamicClockType const&amp;) + 184
5  JavaScriptCore                 0x1951e30d0 WTF::Detail::CallableWrapper&lt;WTF::AutomaticThread::start(WTF::AbstractLocker const&amp;)::$_0, void&gt;::call() + 216
6  JavaScriptCore                 0x195223ac0 WTF::Thread::entryPoint(WTF::Thread::NewThreadContext*) + 180
7  JavaScriptCore                 0x195225c80 WTF::wtfThreadEntryPoint(void*) + 12
8  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
9  libsystem_pthread.dylib        0x18606274c thread_start + 8Heap Helper Thread


Heap Helper Thread
0  libsystem_kernel.dylib         0x18613db58 __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x186057ce4 _pthread_cond_wait$VARIANT$mp + 680
2  JavaScriptCore                 0x195226500 WTF::ThreadCondition::timedWait(WTF::Mutex&amp;, WTF::WallTime) + 152
3  JavaScriptCore                 0x19520b13c WTF::ParkingLot::parkConditionallyImpl(void const*, WTF::ScopedLambda&lt;bool ()&gt; const&amp;, WTF::ScopedLambda&lt;void ()&gt; const&amp;, WTF::TimeWithDynamicClockType const&amp;) + 1908
4  JavaScriptCore                 0x1951e2d18 bool WTF::Condition::waitUntil&lt;WTF::Lock&gt;(WTF::Lock&amp;, WTF::TimeWithDynamicClockType const&amp;) + 184
5  JavaScriptCore                 0x1951e30d0 WTF::Detail::CallableWrapper&lt;WTF::AutomaticThread::start(WTF::AbstractLocker const&amp;)::$_0, void&gt;::call() + 216
6  JavaScriptCore                 0x195223ac0 WTF::Thread::entryPoint(WTF::Thread::NewThreadContext*) + 180
7  JavaScriptCore                 0x195225c80 WTF::wtfThreadEntryPoint(void*) + 12
8  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
9  libsystem_pthread.dylib        0x18606274c thread_start + 8Thread #5


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #6


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #7


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #8


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #9


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #10


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #11


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #12


Thread
0  libsystem_kernel.dylib         0x18611c1ec semaphore_timedwait_trap + 8
1  libdispatch.dylib              0x185fb4d0c _dispatch_sema4_timedwait$VARIANT$mp + 64
2  libdispatch.dylib              0x185fb52a4 _dispatch_semaphore_wait_slow + 72
3  libdispatch.dylib              0x185fc3210 _dispatch_worker_thread + 308
4  libsystem_pthread.dylib        0x18605ed98 _pthread_start + 156
5  libsystem_pthread.dylib        0x18606274c thread_start + 8Thread #13


Thread
0  libsystem_pthread.dylib        0x186062738 start_wqthread + 190Thread #14


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #15


Thread
0  libsystem_kernel.dylib         0x18613e940 __workq_kernreturn + 8
1  libsystem_pthread.dylib        0x18605fbc0 _pthread_wqthread + 348
2  libsystem_pthread.dylib        0x186062740 start_wqthread + 8Thread #16


Thread
0  libsystem_pthread.dylib        0x186062738 start_wqthread + 190Thread #17


Thread
0  libsystem_pthread.dylib        0x186062738 start_wqthread + 190
```

**Comment by SwifterCat** [comment]
@praveek  We have the same issue, an important point I want to mention is that almost all crashes come from iOS 14.0+. We were not able to reproduce this issue for now, it is only be reported in the crash report of our Crash SDK. Let me know if any additional information is needed, and if you require a stack trace of the other threads.
Note: We are using the adobe version, 2.2.6.

**Comment by praveek** [comment]
We have released v2.2.8 which addresses this issue. Let me know if this issue still persists.

**Comment by nasibaliansari** [comment]
@praveek We have the issue in v2.2.8. Please let me know if you need any additional information.
